src/components/capture/Details.tsx

```diff
@@ -46,11 +82,106 @@ function CaptureDetails({ lastPubId, disableLogs }: Props) {
+                getShardDetails(
+                    specs.find(({ spec_type }) => spec_type === entityType)
+                        ?.catalog_name
+                )
+            );
+        }
+    }, [specs, getShardDetails, setShardDetails, entityType]);
+
+    const handlers = {
+        mount: (editor: monacoEditor.editor.IStandaloneCodeEditor) => {
+            editorRef.current = editor;
+        },
+    };
+
     if (pubSpecsError) {
         return <Error error={pubSpecsError} />;
     } else {
         return (
             <Grid container spacing={2}>
+                {shardDetails && (
+                    <>
+                        {shardDetails.errors && (
+                            <Grid item xs={12}>
+                                <Alert
+                                    severity="error"
+                                    sx={{
+                                        '& .MuiAlert-message': {
+                                            width: '100%',
+                                        },
+                                    }}
+                                >
+                                    <Accordion>
+                                        <AccordionSummary
+                                            expandIcon={<ExpandMoreIcon />}
+                                        >
+                                            <Typography>
+                                                <FormattedMessage id="captureDetails.shardDetails.errorTitle" />
+                                            </Typography>
+                                        </AccordionSummary>
+
+                                        <AccordionDetails>
+                                            <Box sx={{ height: 250 }}>
+                                                <Editor
+                                                    defaultLanguage=""
+                                                    theme={
+                                                        theme.palette.mode ===
+                                                        'light'
+                                                            ? 'vs'
+                                                            : 'vs-dark'
+                                                    }
+                                                    options={{
+                                                        lineNumbers: 'off',
+                                                        readOnly: true,
+                                                        scrollBeyondLastLine:
+                                                            false,
+                                                        minimap: {
+                                                            enabled: false,
+                                                        },
+                                                    }}
+                                                    onMount={handlers.mount}
```

Comment:
Don't know what the solution is here... but I feel we are repeating some of the editor stuff. This might be okay because having the react monaco editor already reduces duplication a lot. However, I am thinking we might be able to share the theme and default options somehow.